src/Microsoft.ML.Data/Data/Conversion.cs

```diff
@@ -1164,28 +1164,26 @@ private bool IsStdMissing(ref ReadOnlySpan<char> span)
         /// Utility to assist in parsing key-type values. The min and max values define
         /// the legal input value bounds. The output dst value is "normalized" so min is
         /// mapped to 1, max is mapped to 1 + (max - min).
-        /// Exception is thrown for missing values.
         /// Unparsable or out of range values are mapped to zero with a false return.
         /// </summary>
         public bool TryParseKey(ref TX src, U8 min, U8 max, out U8 dst)
         {
             var span = src.Span;
-            Contracts.Check(span.IsEmpty || !IsStdMissing(ref span), "Missing text value cannot be converted to unsigned integer type.");
-            Contracts.Assert(min <= max);
-
-            // This simply ensures we don't have min == 0 and max == U8.MaxValue. This is illegal since
-            // we map min to 1, which would cause max to overflow to zero. Specifically, it protects
-            // against overflow in the expression uu - min + 1 below.
-            Contracts.Assert((max - min) < U8.MaxValue);
-
             // Both empty and missing map to zero (NA for key values) and that mapping is valid,
             // hence the true return.
-            if (src.IsEmpty)
+            if (src.IsEmpty || IsStdMissing(ref span))
```

Comment:
There shouldn’t be a concept of missing with strings. Missing only applies to floats and double. ROM<char> is a struct so it cannot be null.